_posts/05-06-01-Internationalization-and-Localization.md

```diff
@@ -0,0 +1,348 @@
+
+// here we indicate the default domain the gettext() calls will respond to
+textdomain('main');
+
+// this would look for the string in forum.mo instead of main.mo
+// echo dgettext('forum', 'Welcome back!');
+?>
+{% endhighlight %}
+
+#### 3. Preparing translation for the first run
+To make matters easier - and one of the powerful advantages Gettext has over custom framework i18n packages - is it's custom file type. "Oh man, that's quite hard to understand and edit by hand, a simple array would be easier!" Make no mistake, applications like [Poedit] are here to help - _a lot_. You can get the program from [their website], it's free and available for all platforms. It's a pretty easy tool to get used to, and a very powerful one at the same time - using with responsibility the powers Gettext gave it.
+
+In the first run, you should select "File > New Catalog" from the menu. There you'll have a small screen where we will set the terrain so everything else runs smoothly. You'll be able to find those settings later through "Catalog > Properties":
+
+- Project name and version, Translation Team and email address: useful information that goes in the `.po` file header;
+- Language: here you should use that format we mentioned before, such as `en_US` or `pt_BR`;
+- Charsets: UTF-8, preferably;
+- Source charset: set here the charset used by your PHP files - probably UTF-8 AS well, right?
+- plural forms: here goes those rules we mentioned before - there's a link in there with samples as well;
+- Source paths: here you must include all folders from the project where `gettext()` (and siblings) will happen - this is usually your templates folder(s)
+- Source keywords: this last part is filled by default, but you might need to alter it later - and is one of the powerful points of Gettext. The underlying software knows how the `gettext()` calls look like in several programming languages, but you might as well create your own translation forms. This will be discussed later in the "Tips" section.
+
+After setting those points you'll be prompted to save the file - using that directory structure we mentioned as well, and then it will run a scan through your source files to find the localization calls. They'll be fed empty into the translation table, and you'll start typing in the localized versions of those strings. Save it and a `.mo` file will be (re)compiled into the same folder and ta-dah: your project is internationalized.
+
+#### 4. Translating strings
+As you may have noticed before, there are two main types of localized strings: simple ones and the ones with plural forms. The first ones have simply two boxes: source and localized string. The source string can't be modified as Gettext/Poedit do not include the powers to alter your source files - you should change the source itself and rescan the files. Tip: you may right-click a translation line and it will hint you with the source files and lines where that string is being.  
```

Comment:
> with the source files and lines where that string is being.

I think there's more to this sentence... ?

Or maybe just change `being` to `found` ?